www/app.py

```diff
@@ -2,11 +2,58 @@
 from os import listdir
 from os.path import isfile, join
 import xml.etree.ElementTree
+from flask_socketio import SocketIO, emit
 import Adafruit_GPIO.PWM as pwmLib
+import Adafruit_GPIO.GPIO as gpioLib
 
+# Let SocketIO choose the best async mode
+async_mode = None
 app = Flask(__name__)
+socketio = SocketIO(app, async_mode=async_mode)
+thread = None
 
 pwm = pwmLib.get_platform_pwm(pwmtype="softpwm")
+gpio = gpioLib.get_platform_gpio();
+
+binary_sensors = []
+
+class gpioPoller:
+    def __init__(self, name, pin, rising_event, falling_event):
+        self.name = name
+        self.pin = pin
+        self.rising_event = rising_event
+        self.falling_event = falling_event
+        self.old_val = False
+
+def sensors_thread():
+    while True:
+        global binary_sensors
+        socketio.sleep(0.2)
+        for s in binary_sensors:
+            new_val = gpio.is_high(s.pin)
+            if (s.old_val == False) and (new_val == True):
+                print s.rising_event
+                socketio.emit('binary_sensors', {'data': s.rising_event},
+                    namespace='/api/v1')
+            elif (s.old_val == True) and (new_val == False):
+                print s.falling_event
+                socketio.emit('binary_sensors', {'data': s.falling_event},
+                    namespace='/api/v1')
+            else:
+                pass
+            s.old_val = new_val
+
+@socketio.on('connect', namespace='/api/v1')
+def test_connect():
+    global thread
+    print 'Websocket connected'
+    if thread is None:
+        thread = socketio.start_background_task(target=sensors_thread)
+    emit('status', {'data': 'Connected'})
+
```

Comment:
Actually, I think we should not shut down the thread on disconnect for now. We don't really have session management, so if two people were accessing the API and one left, you wouldn't want to kill the thread for the other person.